### PR TITLE
ValidYoutubeVideo Rule

### PR DESCRIPTION
**Pull request by alasdairgallacher**
Love this package and wanted to make a PR to contribute this.
Here's a feature which makes life so much easier for validating YouTube videos urls.

Example of usage:

`'youtube_video_url' => ['bail', 'required', new ValidYoutubeVideo]`

To prevent unnecessary queries, you can use the `bail` rule.

If the validation rule fails the default error message will be:

> The supplied URL does not look like a Youtube URL.

@alaouy I'm more than happy to spend some time creating the other validation rules for (channel, etc).

### CODE REVIEW

File: src/Rules/ValidYoutubeVideo.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @return bool
+     */
+    public function passes($attribute, $value)
+    {
+        try {
+            $videoId = Youtube::parseVidFromURL($value);
+            $video = Youtube::getVideoInfo($videoId);
```
Comment (Korko) [inline src/Rules/ValidYoutubeVideo.php:31]: You should define the $part argument as ['id'] (cost 0 quota) as you only need to check, not fetch the actual video data.

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,32 @@
+# General
```
Comment (Korko) [inline .gitignore:1]: Not sure why this file is included in the PR.

File: src/Rules/ValidYoutubeVideo.php (hunk 1 above)
Comment (alasdairgallacher) [reply src/Rules/ValidYoutubeVideo.php:31]: Sure thing, will change this

File: .gitignore (hunk 2 above)
Comment (alasdairgallacher) [reply .gitignore:1]: Me neither, will remove this

File: src/Rules/ValidYoutubeVideo.php [hunk 3]
```diff
@@ -28,7 +28,7 @@ public function passes($attribute, $value)
     {
         try {
             $videoId = Youtube::parseVidFromURL($value);
-            $video = Youtube::getVideoInfo($videoId);
+            $video = Youtube::getVideoInfo($videoId)['id'];
```
Comment (Korko) [inline src/Rules/ValidYoutubeVideo.php:31]: No, the idea was to only request the id not to assign only the id
=> Youtube::getVideoInfo($videoId, ['id']);

File: .gitignore [hunk 4]
```diff
@@ -1,5 +0,0 @@
-/vendor
```
Comment (Korko) [inline .gitignore:1]: Now you totally removed the .gitignore in the repository.

File: src/Rules/ValidYoutubeVideo.php (hunk 3 above)
Comment (alasdairgallacher) [reply src/Rules/ValidYoutubeVideo.php]: Sorted this issue, thank you.

File: .gitignore (hunk 4 above)
Comment (alasdairgallacher) [reply .gitignore:1]: @Korko thank you, how do you remove the single file from the pull request?

Comment (Korko) [reply .gitignore:1]: Not sure how using the github interface but in git, should be simple:
`git fetch
git checkout origin/master -- .gitignore`
should reset the file to the master version

Comment (alasdairgallacher) [reply .gitignore]: > Not sure how using the github interface but in git, should be simple:
> `git fetch`
> `git checkout origin/master -- .gitignore`
> should reset the file to the master version

Thank you, this worked and I've pushed to PR

**Comment by Korko** [comment]
Indeed. But careful, you've added also two binary files in the last commit: src/._YoutubeServiceProvider.php and src/Rules/._ValidYoutubeVideo.php not sure it was wanted.

**Comment by alasdairgallacher** [comment]
> Indeed. But careful, you've added also two binary files in the last commit: src/._YoutubeServiceProvider.php and src/Rules/._ValidYoutubeVideo.php not sure it was wanted.

This is not going well haha! Thank you, I think I've sorted it now 🤞

**Comment by alaouy** [comment]
@alasdairgallacher This looks good to me, please just add this to the readme doc, otherwise nobody will know that it exists ; )

@Korko Thank you for the followup

**Comment by alasdairgallacher** [comment]
> @alasdairgallacher This looks good to me, please just add this to the readme doc, otherwise nobody will know that it exists ; )

@alaouy Thank you! I've added to the readme doc under usage section. Please let me know if anything needs changing 👍

**Pull request merged**